From 888b0679045d854c5d6e6f57bc2a954a58dc299b Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Thu, 9 Jul 2020 12:15:12 +0800
Subject: [PATCH] init: Increase the number of threads when initcall async

When initcall_nr_threads=-1, the number of threads set to
num_online_cpus() * 2.

Save boot time about 16ms on rv1126-evb-ddr3-v12 board.

Signed-off-by: Tao Huang <huangtao@rock-chips.com>
Change-Id: I907d03ac21f9e7a74a71ff06385543c0f0fc8c47
---
 init/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/init/main.c b/init/main.c
index 0e4eb2f02d23..f137b7566003 100644
--- a/init/main.c
+++ b/init/main.c
@@ -1123,7 +1123,7 @@ static void __init initcall_init_workers(void)
 	int i;
 
 	if (initcall_nr_workers < 0)
-		initcall_nr_workers = num_online_cpus();
+		initcall_nr_workers = num_online_cpus() * 2;
 
 	if (!initcall_nr_workers)
 		return;
-- 
2.35.3

